### ISSUE TITLE
homebrew_package fails on arm64 macOS w/ chef-workstation + "native" homebrew

### ISSUE THREAD

**Issue by mattlqx** [chef/chef#11606]
## Description
There are a couple issues at play:

1) Presently, Chef Workstation is only available as an x86_64 package so it contains binaries that must be run through Rosetta. When a binary running under Rosetta calls another binary, it too will be run under Rosetta even if the called binary is "fat" and contains a native arm64 version. So when `brew` is shelled out from a chef-client run installed via Chef Workstation, it will be x86_64 arch.

2) Homebrew natively installed on an arm64 Mac will be installed under /opt/homebrew. The path under x86_64 is /usr/local/Homebrew. Homebrew checks the path that `brew` was called from and enforces these paths with errors like:

```
Error: Cannot install in Homebrew on ARM processor in Intel default prefix (/usr/local)!
Please create a new installation in /opt/homebrew using one of the
"Alternative Installs" from:
  https://docs.brew.sh/Installation
```

And vice versa if not providing a /usr/local/bin/brew symlink. (See stacktrace below)

`homebrew_package` fails because of issue 1. `homebrew_tap` and `homebrew_cask` can fail if not provided additional properties because they have a default of `/usr/local/bin/brew` for their homebrew_path attribute and simply symlinking isn't possible because of issue 2 and `homebrew_cask` calls `homebrew_tap` without any properties if the install_cask property is true (and is by default), I think the easy fix there is to just specify the homebrew_path attribute on that to be the same as was specified for it. (Submitted https://github.com/chef/chef/pull/11607 for this)

Issue 1 can be worked around by wrappering `brew` with `arch -64` to select the native 64-bit arch for the machine. At least being able to surface the native architecture in Chef Workstation's x86_64 package would be helpful for determining what actions to take inside the resource (e.g. /usr/local/bin/brew vs. /opt/homebrew/bin/brew). You can use arch with uname to accurately ascertain the native architecture like `arch -64 uname -m`. Maybe a helper inside Chef could handle this or be a part of ohai?

## Chef Version
➜ chef --version                             
Chef Workstation version: 21.5.420
Chef Infra Client version: 17.0.242
Chef InSpec version: 4.36.4
Chef CLI version: 5.0.1
Chef Habitat version: 1.6.288
Test Kitchen version: 2.11.2
Cookstyle version: 7.10.1

## Platform Version
macOS 11.3.1 on MacBook Pro M1

## Replication Case

- Install "native" homebrew via brew.sh standard install.
- Install Chef Workstation from dmg.
- Run a recipe with `homebrew_package` like

```
  homebrew_package 'coreutils' do
    action :upgrade
  end
```

## Client Output
See stacktrace.

## Stacktrace

```
* homebrew_package[coreutils] action upgrade[2021-05-20T15:40:14-07:00] INFO: Processing homebrew_package[coreutils] action upgrade (macos-workstation::base_software line 23)

    
    ================================================================================
    Error executing action `upgrade` on resource 'homebrew_package[coreutils]'
    ================================================================================
    
    Mixlib::ShellOut::ShellCommandFailed
    ------------------------------------
    Expected process to exit with [0], but received '1'
    ---- Begin output of ["brew", "install", "coreutils"] ----
    STDOUT: 
    STDERR: Error: Cannot install under Rosetta 2 in ARM default prefix (/opt/homebrew)!
    To rerun under ARM use:
        arch -arm64 brew install ...
    To install under x86_64, install Homebrew into /usr/local.
    ---- End output of ["brew", "install", "coreutils"] ----
    Ran ["brew", "install", "coreutils"] returned 1
    
    Resource Declaration:
    ---------------------
    # In /Users/matt/.chef/local-mode-cache/cache/cookbooks/macos-workstation/recipes/base_software.rb
    
     23:   homebrew_package pkg do
     24:     action :upgrade
     25:   end
     26: end
    
    Compiled Resource:
    ------------------
    # Declared in /Users/matt/.chef/local-mode-cache/cache/cookbooks/macos-workstation/recipes/base_software.rb:23:in `block in from_file'
    
    homebrew_package("coreutils") do
      package_name "coreutils"
      action [:upgrade]
      default_guard_interpreter :default
      declared_type :homebrew_package
      cookbook_name "macos-workstation"
      recipe_name "base_software"
    end
    
    System Info:
    ------------
    chef_version=17.0.242
    platform=mac_os_x
    platform_version=11.3.1
    ruby=ruby 2.7.3p183 (2021-04-05 revision 6847ee089d) [x86_64-darwin18]
    program_name=/usr/local/bin/chef-client
    executable=/opt/chef-workstation/bin/chef-client
```

**Comment by lamont-granquist** [comment]
I mean the fix for #1 is to get workstation compiled natively for M1 mac, we're not going to patch things so that rosetta-run chef will be able to install native homebrew into /opt/homebrew.  And that needs to be addressed in chef-workstation (I know that's in flight, but have no idea when that might be delivered, but its pretty high priority, just a large yak shave to get everything building there).

The /usr/local -> /opt/homebrew change is something that we should do though so that native chef-client does the right thing there.

**Comment by tas50** [comment]
I'm going to decline this one in Chef Infra Client since the fix is actually to get Chef Workstation building on arm for macOS. There's a plan to do this once we get habitat building on arm.
